aptos-node/src/storage.rs

```diff
@@ -18,7 +18,18 @@ pub(crate) fn maybe_apply_genesis(
     db_rw: &DbReaderWriter,
     node_config: &NodeConfig,
 ) -> Result<Option<LedgerInfoWithSignatures>> {
-    let genesis_waypoint = node_config.base.waypoint.genesis_waypoint();
+    // We read from the storage genesis waypoint and fallback to the node config one if it is none
+    let genesis_waypoint = node_config
+        .execution
+        .genesis_waypoint_config
```

Comment:
why not just `execution.genesis_waypoint`...? (more symmetric to existing `base.waypoint`)